<?xml version="1.0" encoding="utf-8"?>
<launch>

    <!-- Define the paths to the parameter files -->
    <arg name="topics_param_file_name" default="$(find state_controller)/params/topics.yaml"/>
    <arg name="game_master_param_file_name" default="$(find state_controller)/params/game_master.yaml"/>
    
    <param name="~bot_type" value="quackman" />
    <param name="vehicle_name" value="$(arg vehicle_name)" />

    <!-- Launch the node, loading the specified parameter files (.yaml) -->
    <node name="game_master_connector" pkg="state_controller" type="game_master_connector.py" output="screen" required="true">
        <rosparam command="load" file="$(arg topics_param_file_name)" ns="topics"/>
        <rosparam command="load" file="$(arg game_master_param_file_name)" ns="game_master"/>
        <!-- <rosparam name="bot_type" value="quackman" /> -->
    </node>
    
</launch>